docs/section5/Rule5-8.md

```diff
@@ -0,0 +1,45 @@
+
+# Envelope - Rule 5-8  
+
+**Rule ID:** 5-8  
+**Rule Description:**  Fenestration U-factors for new buildings, existing buildings, and additions shall match the appropriate requirements in Tables G3.4-1 through G3.4-8 for the applicable glazing percentage for Uall.  
+**Rule Assertion:** Baseline RMR = expected value  
+**Appendix G Section:** Section G3.1-5(d) Building Envelope Modeling Requirements for the Baseline building  
+**Appendix G Section Reference:** Tables G3.4-1 to G3.4-8  
+
+**Applicability:** All required data elements exist for B_RMR  
+**Applicability Checks:**  None  
+
+**Manual Check:** None  
+**Evaluation Context:** Each Data Element  
+**Data Lookup:** Tables G3.4-1 to G3.4-8  
+
+## Rule Logic:  
+
+- Get building climate zone: ```climate_zone = B_RMR.weather.climate_zone```  
+
+- For each building segment in the Baseline model: ```for building_segment_baseline in B_RMR.building.building_segments:```  
+
+  - For each thermal_block in building segment: ```for thermal_block_baseline in building_segment_baseline.thermal_blocks:```  
+
+  - For each zone in thermal block: ```for zone_baseline in thermal_block_baseline.zones:```  
+
+  - For each space in thermal zone: ```for space_baseline in zone_baseline.spaces:```  
+
+    - Get surfaces in space: ```for surface_baseline in space_baseline.surfaces:```  
+
+      - Check if surface is exterior wall: ```if ( surface_baseline.classification == "WALL" ) AND ( surface_baseline.adjacent_to == "AMBIENT" ): wall_surface_baseline = surface_baseline```  
+
+        - Calculate the total wall area: ```total_wall_area_baseline += wall_surface_baseline.area```  
+
+        - Get the U-factor for vertical glazings: ```window_performance_value_baseline.append(window.u_factor for window in wall_surface_baseline.fenestration_subsurfaces)```  
+
```

Comment:
I believe there needs to be a step before this which finds all fenestration surfaces on a wall surface and determines the group of 'windows' (doors are not included). It seems to be implied in 'Get the U-factor for vertical glazings' but would be good to explicitly state it as well
@glazer, i believe 90.1 identifies doors which are >50% glazing as windows. I believe the 'Fenestration Classification' takes care of, correct?

Reply:
@supriyagoel Fenestration Classification only has options for Window, Skylight, Door. It does not really talk about doors with over 50% glass. We don't have enough descriptors in the schema to determine that right now. Maybe we should have two areas in the Fenestration data group for areas where light can pass through or not. Or else another classification for "GlazedDoor." What do you think?

Reply:
I think classification that we currently have (window, skylight, door) is sufficient if we add provisions for each category to have both opaque part and glazed part. This would capture doors but also windows/skylights with frame and glazing portions explicitly modeled.

Reply:
@supriyagoel currently the fenestration object has a single set of properties - area, U-factor, SHGC, etc. We could potentially modify it to allow two sets - one for opaque segment and another for transparent. Or perhaps U-factor and area would the the only two properties allowed for opaque part. Then, if fenestration object is a door, RCT can compare glazed area to opaque area to determine applicability of App G rules.

Reply:
I agree, I will propose the fenestration data group have: glazed_area, opaque_area, glazed_u_factor, opaque_u_factor.